### PR TITLE
Wrapper Update

### PR DESCRIPTION
**Pull request by shazbert**
# Description

Deposit address update for funding account.

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

go test ./...

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool 
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/binance/binance.go [hunk 1]
```diff
@@ -603,17 +604,36 @@ func (b *Binance) SendAuthHTTPRequest(method, path string, params url.Values, re
...
 	headers := make(map[string]string)
 	headers["X-MBX-APIKEY"] = b.APIKey
 
 	if b.Verbose {
-		log.Debugf("sent path: \n%s\n", path)
+		log.Debugf("sent path: %s", path)
 	}
+
 	path = common.EncodeURLValues(path, params)
+	path += fmt.Sprintf("&signature=%s", hmacSignedStr)
```
Comment (thrasher-) [inline exchanges/binance/binance.go]: Guessing this is because alphabetical sorting breaks the request?

File: exchanges/binance/binance_test.go [hunk 2]
```diff
@@ -491,3 +492,18 @@ func TestWithdrawInternationalBank(t *testing.T) {
 		t.Errorf("Expected '%v', recieved: '%v'", common.ErrFunctionNotSupported, err)
 	}
 }
+
+func TestGetDepositAddress(t *testing.T) {
+	b.Verbose = true
+	if testAPIKey != "" && testAPISecret != "" {
```
Comment (thrasher-) [inline exchanges/binance/binance_test.go]: Can use areTestAPIKeysSet()

File: exchanges/binance/binance_test.go [hunk 3]
```diff
@@ -423,9 +422,11 @@ func TestGetAccountInfo(t *testing.T) {
 		t.Skip()
 	}
 
+	b.Verbose = true
```
Comment (thrasher-) [inline exchanges/binance/binance_test.go]: Remove

File: exchanges/zb/zb.go [hunk 4]
```diff
@@ -9,6 +9,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/thrasher-/gocryptotrader/currency/pair"
+
```
Comment (thrasher-) [inline exchanges/zb/zb.go]: remove newline

File: exchanges/binance/binance.go (hunk 1 above)
Comment (shazbert) [reply exchanges/binance/binance.go]: That's correct

Comment (shazbert) [reply exchanges/binance/binance.go]: Signature needs to be put at the end.

File: exchanges/poloniex/poloniex_wrapper.go [hunk 5]
```diff
@@ -236,7 +244,18 @@ func (p *Poloniex) GetOrderInfo(orderID int64) (exchange.OrderDetail, error) {
...
 func (p *Poloniex) GetDepositAddress(cryptocurrency pair.CurrencyItem) (string, error) {
-	return "", common.ErrNotYetImplemented
+	a, err := p.GetDepositAddresses()
+	if err != nil {
+		return "", err
+	}
+
+	address, ok := a.Addresses[cryptocurrency.Upper().String()]
+	if !ok {
+		return "", fmt.Errorf("Cannot find deposit addres for %s",
```
Comment (gloriousCode) [inline exchanges/poloniex/poloniex_wrapper.go]: Typo with address

File: exchanges/huobi/huobi_wrapper.go [hunk 6]
```diff
@@ -149,76 +149,87 @@ func (h *HUOBI) UpdateOrderbook(p pair.CurrencyPair, assetType string) (orderboo
...
-	}
+		acc.ID = strconv.FormatInt(account.ID, 10)
 
-	var currencyData = make(map[string]*hold)
-	for _, data := range acc {
-		_, ok := currencyData[data.Currency]
-		if !ok {
-			currencyData[data.Currency] = &hold{}
+		if account.State == "working" {
+			acc.Working = true
```
Comment (gloriousCode) [inline exchanges/huobi/huobi_wrapper.go]: This is the only place I see `.Working` having the ability to be `false`. However, I don't see anything that changes behaviour for when the account is false. Would you be able to explain what the working is used for?

File: exchanges/bitfinex/bitfinex_wrapper.go [hunk 7]
```diff
@@ -230,7 +237,7 @@ func (b *Bitfinex) GetOrderInfo(orderID int64) (exchange.OrderDetail, error) {
 
 // GetDepositAddress returns a deposit address for a specified currency
 func (b *Bitfinex) GetDepositAddress(cryptocurrency pair.CurrencyItem) (string, error) {
-	return "", common.ErrNotYetImplemented
+	return "", common.ErrFunctionNotSupported
```
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/bitfinex/bitfinex_wrapper.go]: I was looking for that for ages, :S THANKS! Maybe I was looking at v2?

File: exchanges/anx/anx_wrapper.go [hunk 8]
```diff
@@ -306,7 +310,7 @@ func (a *ANX) GetOrderInfo(orderID int64) (exchange.OrderDetail, error) {
 
 // GetDepositAddress returns a deposit address for a specified currency
 func (a *ANX) GetDepositAddress(cryptocurrency pair.CurrencyItem) (string, error) {
-	return "", common.ErrNotYetImplemented
+	return "", common.ErrFunctionNotSupported
```
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/anx/anx_wrapper.go]: yes :D

File: exchanges/bitmex/bitmex_test.go [hunk 9]
```diff
@@ -638,3 +637,22 @@ func TestWithdrawInternationalBank(t *testing.T) {
 		t.Errorf("Expected '%v', recieved: '%v'", common.ErrFunctionNotSupported, err)
 	}
 }
+
+func TestGetDepositAddress(t *testing.T) {
+	_, err := b.GetDepositAddress("Bro")
```
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/bitmex/bitmex_test.go]: I will change that.

File: exchanges/huobihadax/huobihadax_wrapper.go [hunk 10]
```diff
@@ -296,7 +307,7 @@ func (h *HUOBIHADAX) GetOrderInfo(orderID int64) (exchange.OrderDetail, error) {
 
 // GetDepositAddress returns a deposit address for a specified currency
 func (h *HUOBIHADAX) GetDepositAddress(cryptocurrency pair.CurrencyItem) (string, error) {
-	return "", common.ErrNotYetImplemented
+	return "", common.ErrFunctionNotSupported
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/huobihadax/huobihadax_wrapper.go]: Actually, sorry about that, this seems to be for retrieving previous withdrawal/deposits (which so happen to contain addresses). I'll resolve these

File: exchanges/bitstamp/bitstamp.go [hunk 11]
```diff
@@ -565,29 +567,34 @@ func (b *Bitstamp) OpenInternationalBankWithdrawal(amount float64, currency,
...
-// crypto - example "btc", "ltc", "eth", or "xrp"
+// crypto - example "btc", "ltc", "eth", "xrp" or "bch"
 func (b *Bitstamp) GetCryptoDepositAddress(crypto string) (string, error) {
-	type response struct {
-		Address string `json:"address"`
-	}
-	resp := response{}
+	var resp string
 
 	switch common.StringToLower(crypto) {
```
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/bitstamp/bitstamp.go]: yeah sure

[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/bitstamp/bitstamp.go]: Yeah sure.

File: exchanges/bitmex/bitmex_test.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/bitmex/bitmex_test.go]: No! I was saying its good that you did it! 😄

File: exchanges/bitstamp/bitstamp.go [hunk 12]
```diff
@@ -663,5 +670,23 @@ func (b *Bitstamp) SendAuthenticatedHTTPRequest(path string, v2 bool, values url
...
+	errCap := struct {
+		Error string `json:"error"`
+	}{}
+
+	err := b.SendPayload("POST", path, headers, readerValues, &interim, true, b.Verbose)
+	if err != nil {
+		return err
+	}
+
+	if err := common.JSONDecode(interim, &errCap); err == nil {
```
Comment (gloriousCode) [inline exchanges/bitstamp/bitstamp.go]: Does this mean that when JSONDecode errors into err, it is uncaught?

Comment (shazbert) [reply exchanges/bitstamp/bitstamp.go]: Yeah if it errors we do not need to handle that because of the secondary unmarshal.

**Comment by thrasher-** [comment]
🏆 !

**Pull request merged**